recommendations/local-resource-conservation.md

```diff
@@ -0,0 +1,334 @@
+  0.5).
+
+A node implementing resource bucketing limits exposure on its outgoing channel:
+* MUST choose `protected_slot_count` <= the remote channel peer's 
+  `max_accepted_htlcs`.
+* MUST choose `protected_liquidity_portion` in [0;1].
+
+For each `update_add_htlc` proposed by an incoming channel:
+* If `endorsed` is non-zero AND the incoming channel has sufficient local 
+  reputation for the HTLC (see [Recommendations for Reputation Scoring](#recommendations-for-reputation-scoring)):
+  * SHOULD forward the HTLC as usual.
+  * SHOULD set `endorsed` to 1 in the outgoing `update_add_htlc`.
+* Otherwise: 
+  * SHOULD reduce the remote peer's `max_accepted_htlcs` by 
+    `protected_slot_count` for the purposes of the proposed HTLC.
+  * SHOULD reduce the `max_htlc_value_in_flight` by 
+    `protected_liquidity_portion` * `max_htlc_value_in_flight`.
+  * SHOULD set `endorsed` to `0` in the outgoing `update_add_htlc`.
+
+## Implementation Notes
+
+### Decaying Average
+Rolling windows specified in this write up may be implemented as a decaying 
+average to minimize the amount of data that needs to be stored per-channel. In 
+flight HTLCs can be accounted for separately to this calculation, as the node 
+will already have data for these HTLCs available.
+
+Track the following values for each rolling window: 
+* `last_update`: stores the timestamp of the last update to the decaying 
+    average, expressed in seconds.
+* `decaying_average`: stores the value of the decaying average.
+* `decay_rate`: a constant rate of decay based on the rolling window chosen, 
+  calculated as: `((1/2)^(2/window_length_seconds))`.
+
+To update the `decaying_average` at time `t`:
+* `last_update_diff` = `now` - `last_update`.
+* `decaying_average` = `decaying_average` * `decay_rate` ^ `last_update_diff`.
+* `last_update` = `t`.
+
+When assessing whether a peer has sufficient reputation for a HTLC: 
+* MUST update `decaying_average` as described above.
+
+When updating the `decaying_average` to add the `effective_fee` of a newly 
+resolved HTLC at timestamp `t`: 
+* MUST update `decaying_average` as described above.
+* `decaying_average` = `decaying_average` + `effective_fee`
+
+### Bootstrapping Outgoing Channel Revenue
+New channels with no revenue history: 
+* MAY choose not to endorse any HTLCs in their first two weeks of operation 
+  to establish baseline revenue.
+* MAY choose to use `outgoing_channel_revenue` for a similar channel as a 
+  default starting point for scoring reputation.
+
+### Multiple Channels
+If the local node has multiple channels open with the incoming and outgoing 
+peers:
+* MAY consider `incoming_channel_revenue` across all channels with the peer 
+  when assessing reputation.
+* MAY consider `outgoing_channel_revenue` for all channels with the outgoing 
+  peer, but SHOULD take care to [bootstrap](#bootstrapping-outgoing-channel-revenue)
+  new channels so they do not lower the reputation threshold for existing ones.
```

Comment:
> * MAY consider `incoming_channel_revenue` across all channels with the peer 
> when assessing reputation.
> * MAY consider `outgoing_channel_revenue` for all channels with the outgoing 
>  peer, but SHOULD take care to [bootstrap](#bootstrapping-outgoing-channel-revenue)

Minor - ....Hmmmmmm, it could be interesting to adopt the communalized reputation assesment
over many channels from a upstream peer on the HTLC timing. Given their lower and
upper bounded by the `height_added` and `resolution_window`, a set of slow-jamming
HTLCs might have to fit within the same window.

Especially, it could be useful to prevent sudden spikes of slow-jamming HTLCs to
occupy liquidity / slots, when those slow-jamming are triggered with few hops of
depths in the graph. While not downgrading the forwarding of the upstream peers
the rest of the time. It could be a thing.